packages/runtime/src/templates/edge-shared/utils.ts

```diff
@@ -250,6 +266,12 @@ export const buildResponse = async ({
     res.headers.set('x-nextjs-redirect', relativizeURL(redirect, request.url))
   }
 
+  const nextRedirect = res.headers.get('x-nextjs-redirect')
```

Comment:
We tested this together and it works well. I still have no idea why we have to set the header first, get it and the reset it. If anyone knows why, I'm curious. 🕵🏻